src/barfturn.ts

```diff
@@ -0,0 +1,270 @@
+import {
+  availableAmount,
+  canAdventure,
+  currentRound,
+  inebrietyLimit,
+  myAdventures,
+  myInebriety,
+  myLevel,
+  runChoice,
+  runCombat,
+  totalTurnsPlayed,
+  toUrl,
+  use,
+  useFamiliar,
+  visitUrl,
+} from "kolmafia";
+import {
+  $effect,
+  $item,
+  $items,
+  $location,
+  $monster,
+  $skill,
+  adventureMacro,
+  adventureMacroAuto,
+  clamp,
+  Counter,
+  ensureEffect,
+  get,
+  getModifier,
+  have,
+  Requirement,
+  SourceTerminal,
+} from "libram";
+import { Macro, withMacro } from "./combat";
+import { completeBarfQuest } from "./dailies";
+import { estimatedTurns } from "./embezzler";
+import { barfFamiliar, freeFightFamiliar, meatFamiliar } from "./familiar";
+import {
+  embezzlerLog,
+  globalOptions,
+  kramcoGuaranteed,
+  safeRestore,
+  setChoice,
+  steveAdventures,
+} from "./lib";
+import { meatMood } from "./mood";
+import {
+  familiarWaterBreathingEquipment,
+  freeFightOutfit,
+  meatOutfit,
+  tryFillLatte,
+  waterBreathingEquipment,
+} from "./outfit";
+import { determineDraggableZoneAndEnsureAccess, digitizedMonstersRemaining } from "./wanderer";
+const embezzler = $monster`Knob Goblin Embezzler`;
+
+function embezzlerPrep(requirements?: Requirement) {
+  useFamiliar(meatFamiliar());
+  meatOutfit(true, requirements);
+}
+
+function freeFightPrep(requirements?: Requirement) {
+  useFamiliar(freeFightFamiliar(true));
+  freeFightOutfit(requirements);
+}
+
+const sober = () => myInebriety() <= inebrietyLimit();
+
+type Turn = {
+  name: string;
+  available: () => boolean;
+  perform: () => boolean;
+};
+
+const turns: Turn[] = [
+  {
+    name: "Lights Out",
+    available: () =>
+      totalTurnsPlayed() % 37 === 0 && totalTurnsPlayed() !== get("lastLightsOutTurn"),
+    perform: () => {
+      const steveRoom = get("nextSpookyravenStephenRoom");
```

Comment:
have we tested whether or not names cause any issues w/ using this pref this way? ensured the pref's wording doesn't like, have an extra the or removes a the etc? i would imagine we have tested this but i must ask.

Reply:
Garbo has been doing this since May 22nd.